### PR TITLE
WIP: Searchable gutenberg icon library

### PR DESCRIPTION
**Pull request by bacoords**
I've added a link to the homepage, but everything lives on /icons. VERY open to feedback on this (I have a few thoughts already)

### CODE REVIEW

**Comment by aurooba** [comment]
Most of the time, I personally use/need just the icon name. So perhaps we need the option to copy just the name from the icon card, and then maybe an auto-update on select area near the top where you can copy a working example with your chosen icon?

**Comment by bacoords** [comment]
Yeah there's a few issues around- mostly other icon libraries like this show the full icon info on a separate page, because it's too much code to fit in the small grid (i.e. all icons https://icons.getbootstrap.com/ vs a single icon https://icons.getbootstrap.com/icons/0-circle/) which I think is a better approach but wanted to check with you first. I'd like to have each icon open to a new page that lets you copy each piece separately but also shows ALL the code, not just the name.

**Comment by aurooba** [comment]
As we discussed in our call – I think it makes sense to do a combination. From the grid itself, it would be lovely if you could quickly copy the name – since that's often what I'm looking for and need. But then clicking on the icon itself could open up it's own page with the full code for maybe two examples:

1. using it with the Icon component like you have now
2. using it inside a button or panel component, which is a fairly common use of the icons

I also like the idea of having just the SVG for copy/pasting purposes like how Bootstrap has it, to make it super easy for front-end/non-block-editor use so you don't have to import the entire icon package if you don't want to.

File: styles/_icons.scss [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	gap: 1rem;
+	max-width: 1080px;
+	margin:0 auto;
+
+	padding:1rem;
+	@media screen and (min-width: 768px) {
+		grid-template-columns: repeat(3, minmax(0, 1fr));
+	}
+	@media screen and (min-width: 1080px) {
+		grid-template-columns: repeat(5, minmax(0, 1fr));
```
Comment (aurooba) [inline styles/_icons.scss:35]: I think at most, we should have 4 columns, since some of the names are kind of long

File: components/Footer.js [hunk 2]
```diff
@@ -7,6 +7,10 @@ export default function Footer() {
 					<a href="https://aurooba.com" target="_blank" rel="noopener">
 						Aurooba
 					</a>
+					{" "}and{" "}
```
Comment (aurooba) [inline components/Footer.js:10]: This feels a bit disjointed, could we improve this so the spaces aren't separate from the `and`?

File: pages/icons.js
Comment (aurooba) [inline pages/icons.js:1]: It's probably time to pull out the header as a component of its own, among other things. We can definitely do that in a separate PR, but just noting this here.

**Review by aurooba — CHANGES_REQUESTED** [review]
Overall this is so awesome. I'm looking forward to merging this! :D

File: styles/_icons.scss (hunk 1 above)
Comment (bacoords) [reply styles/_icons.scss:35]: What about a smaller font size?

File: pages/icons.js
Comment (bacoords) [reply pages/icons.js:1]: Created a new issue #3

**Comment by aurooba** [comment]
Just flagging that the new header isn't responsive (but looks fantastic on desktop!)

**Review by aurooba — APPROVED** [review]
Looks lovely! @bacoords you're clear to merge!
